Adding click to handle arguments + small changes

arodrime opened the pull request:
Hello, 

First of all, thanks for this work. I'll use it.

But before I made some changes to your script, mostly adding `Click` library to handle options. I'm sharing them so other people can make use it and with the hope that you will like the changes =).

I also:
- allowed the script to look in multiple zones to improve chances to get a server.
- changed the print a bit (using .format() and debug level `INFO` to see what's happening) 
- some more small stuff, including increasing versions dependency (selenium was not compatible with current firefox for example).

arodrime committed:
Adding click to handle arguments + small changes

arodrime committed:
Adding more options - payment frequency/number of servers

arodrime committed:
Fix contract checkboxes and validation + read timeout

arodrime commented:
- Added payment and quantity option.
 - Configured default read timeout to handle disconnection / networks switch
- Fixed contract checkboxes and validation button selectors (according to ks webpage early 2019)

arodrime committed:
Fixed log path / screen names

PLNech commented:
Hi @arodrime, sorry I took so long to reply! 

Thanks for the kind words and for sharing these improvements :)

As soon as I can review this I'll merge your branch, and in the meantime will mention it to other interested users 🚀

arodrime committed:
Update November 2020 - popup / select payment

emirdora commented:
@PLNech I added today:

* Handling the popup for cookies.
* Handling selecting payment method
* Updated API link to check availability

This script should work.

Tested with:

```
python3 buyKim.py --ovh-user <myemail> --ovh-pass <mypass> --ref-product 1801sk14 --debug
```

* replace "1801sk14" by any available server for testing. 
* To really buy, remove the --debug option and pick the desired server.

arodrime commented:
@PLNech I added today:

* Handling the popup for cookies.
* Handling selecting payment method
* Updated API link to check availability

This script should work.

Tested with:

```
python3 buyKim.py --ovh-user <myemail> --ovh-pass <mypass> --ref-product 1801sk14 --debug
```

* replace "1801sk14" by any available server for testing. 
* To really buy, remove the --debug option and pick the desired server.

More options?

```
BuyKim$ python3 buyKim.py --help
Usage: buyKim.py [OPTIONS]

Options:
  -t, --timeout-conn INTEGER      Maximum time in seconds to wait for
                                  webservice answer.  [default: 5]
  -i, --interval FLOAT            Minimum interval in seconds between two
                                  requests  [default: 7.5]
  --product-family TEXT           The family of servers (ie. "Kimsufi"/"So you
                                  Start")  [default: Kimsufi]
  -p, --ref-product TEXT          Reference of the server (ie 1801sk12 for
                                  KS1, 1801sys29 for some soYouStart servers
                                  [default: 1801sk12]
  -z, --ref-zones TEXT            Data center short name(s) (ie "-z gra -z
                                  rbx")  [default: gra, rbx, lon, fra]
  -q, --quantity INTEGER          Number of servers to rent - 1 to 5 (Maximum)
                                  [default: 1]
  -f, --payment-frequency INTEGER
                                  Receive the bill every 1,3,6 or 12 month
                                  [default: 1]
  --ovh-user TEXT
  --ovh-pass TEXT
  --debug / --no-debug            Debug mode, disable by default. Add --debug
                                  flag to enable
  --help                          Show this message and exit.
```

piratack007 commented:
> @PLNech I added today:
> 
> * Handling the popup for cookies.
> * Handling selecting payment method
> * Updated API link to check availability
> 
> This script should work.
> 
> Tested with:
> 
> ```
> python3 buyKim.py --ovh-user <myemail> --ovh-pass <mypass> --ref-product 1801sk14 --debug
> ```
> 
> * replace "1801sk14" by any available server for testing.
> * To really buy, remove the --debug option and pick the desired server.
> 
> More options?
> 
> ```
> BuyKim$ python3 buyKim.py --help
> Usage: buyKim.py [OPTIONS]
> 
> Options:
>   -t, --timeout-conn INTEGER      Maximum time in seconds to wait for
>                                   webservice answer.  [default: 5]
>   -i, --interval FLOAT            Minimum interval in seconds between two
>                                   requests  [default: 7.5]
>   --product-family TEXT           The family of servers (ie. "Kimsufi"/"So you
>                                   Start")  [default: Kimsufi]
>   -p, --ref-product TEXT          Reference of the server (ie 1801sk12 for
>                                   KS1, 1801sys29 for some soYouStart servers
>                                   [default: 1801sk12]
>   -z, --ref-zones TEXT            Data center short name(s) (ie "-z gra -z
>                                   rbx")  [default: gra, rbx, lon, fra]
>   -q, --quantity INTEGER          Number of servers to rent - 1 to 5 (Maximum)
>                                   [default: 1]
>   -f, --payment-frequency INTEGER
>                                   Receive the bill every 1,3,6 or 12 month
>                                   [default: 1]
>   --ovh-user TEXT
>   --ovh-pass TEXT
>   --debug / --no-debug            Debug mode, disable by default. Add --debug
>                                   flag to enable
>   --help                          Show this message and exit.
> ```

hi thanks a lot for your contribution sir, please where can i find the new script?? or must i download https://github.com/PLNech/BuyKim/blob/master/buyKim.py  ???
regards

piratack007 commented:
Thnaks a lot 🥇

PLNech commented:
Hey @arodrime, thanks that's great! I'll take some time during an evening this week to test it and finally merge this :tada:
